pkg/providers/azure.go

```diff
@@ -165,16 +166,24 @@ func (sc *SetUpCmd) CreateServicePrincipal() error {
 }
 
 func (sc *SetUpCmd) assignSpRole(ctx context.Context) error {
-	log.Debug("Assigning contributor role to service principal...")
+	if err := sc.assignRole(ctx, "b24988ac-6180-42a0-ab88-20f7382dd24c"); err != nil { // Contributor role ID
```

Comment:
We can avoid having this duplicate by declaring the roles into a var of string slice like
```go
var spRoles := []string {
  "b24988...", // contributor
  "5af6afb...", // Azure Kubernetes Fleet Writer
}
``` 
and then do
```go 
for _, role := range spRoles {
  if err := sc.assignRole(ctx, role); err != nil {
    return err
  }
}
```